src/Serilog.Extensions.Logging.File/Serilog.Extensions.Logging.File.csproj

```diff
@@ -32,9 +32,9 @@
   <ItemGroup>
     <PackageReference Include="Serilog" Version="2.10.0" />
     <PackageReference Include="Serilog.Sinks.Async" Version="1.5.0" />
-    <PackageReference Include="Serilog.Sinks.RollingFile" Version="3.3.0" />
     <PackageReference Include="Serilog.Extensions.Logging" Version="3.1.0" />
     <PackageReference Include="Serilog.Formatting.Compact" Version="1.1.0" />
+    <PackageReference Include="Serilog.Sinks.File" Version="5.0.0" />
```

Comment:
It's not possible to migrate to _Serilog.Sinks.File_ without a substantial breaking change because of the differences in path formatting and rolling options; I think it's time we migrated, but there's more to discuss around it, so it'd be best discussed in a separate PR.

Reply:
so I do not need to migrate in this pr? I do not see any option in old library about it, https://github.com/serilog/serilog-sinks-rollingfile/blob/dev/src/Serilog.Sinks.RollingFile/RollingFileLoggerConfigurationExtensions.cs and without option rotation will not work

Reply:
Hi @kgrodimov - no need to migrate in this PR. The old library determines the rolling interval from the output path format string - if it includes `{Date}` then it rolls by day, etc.

Reply:
so how can I add roll by size if old library does not support this parameter

Reply:
Ah, sorry - I see the problem now, sorry about that!

To move forward I think we'll need a plan for dealing with the incompatibilities between the old and new sinks.

My preference would be to detect format placeholders in the path string, and either:

 * Remove them if they're in the exact form and position that the _File_ sink would satisfy, e.g. `log-{Date}.txt` can be converted into `log-.txt` and _Serilog.Sinks.File_ will match the old behavior reasonably well, or
 * Throw a configuration-time exception reporting that the format is unsupported (which should be enough for most users to pick it up on first run, at development time, on upgrading the package)

https://github.com/serilog/serilog-sinks-rollingfile#filename-format-specifiers has details on the old sink's path formatting system.

The docs etc. will also need going over.

I realize that this expands the task quite a lot so please don't feel obligated to take all of this on :-) Thanks for your efforts to date 👍

Reply:
I think the `log-{Date}.txt` will cover the overwhelming majority of cases our there, so it'd be nice to keep accepting that and just trim out the `{Date}` to make the file paths compatible with the new implementation. We might end up with a flood of breakage/support otherwise.

This only works when `{Date}` immediately precedes the extension. If any other specifiers are there, or `{Date}` appears elsewhere in the path, we'd want to throw an exception.

If you're keen to explore it, we could also consider adding a `RollingInterval` argument and a `shared` flag, which would then cover 99% of the use cases for _File_.

HTH!

Reply:
So just trim out rollingInterval only if it immediately precedes the extension and pass correct interval with RollingInterval to [Serilog.Sinks.File](https://github.com/serilog/serilog-sinks-file/tree/2d2e00b31c4c1ed36dbbca8f4a43389856964fae/src/Serilog.Sinks.File) according to parsed value. So not only {Date}, but {Hour}, {HalfHour} also, or just {Day} as most popular. 

New library does not support {HalfHour} btw, https://github.com/serilog/serilog-sinks-file/blob/2d2e00b31c4c1ed36dbbca8f4a43389856964fae/src/Serilog.Sinks.File/Sinks/File/RollingIntervalExtensions.cs#L23. So for {HalfHour} can I throw exception?

Reply:
Sorry about the slow reply; yes, I think throwing an exception in that case would make sense 👍